airsonic-main/src/main/java/org/airsonic/player/controller/DownloadController.java

```diff
@@ -103,7 +103,7 @@
 
     @GetMapping
     public ResponseEntity<Resource> handleRequest(Principal p,
-            @RequestParam Optional<Integer> id,
+            @RequestParam(name = "id") Optional<List<Integer>> ids,
```

Comment:
I changed this to a List but still using the Optional. But I'm not quiet sure if I could change this to (required = false) and check for null instead like the other parameters. what's the difference?

Reply:
No difference. Just in the coding style that you want to adopt or pursue later on in the method (imperative or functional, or even declarative). Monads like the Optional help you pursue the latter, required = false are more aligned to the former.